src/main/java/duke/dependencies/parser/Controller.java

```diff
@@ -0,0 +1,148 @@
+package duke.dependencies.parser;
+
+import duke.dependencies.dukeexceptions.DukeException;
+import duke.dependencies.dukeexceptions.EmptyTaskException;
+import duke.dependencies.dukeexceptions.InvalidDateException;
+import duke.dependencies.dukeexceptions.UnknownCommandException;
+import duke.dependencies.dukeexceptions.UnspecifiedDateException;
+import duke.dependencies.executable.Executable;
+import duke.dependencies.executor.Executor;
+import duke.dependencies.storage.Storage;
+
+
+/**
+ * Parser class which parses the given input. Checks if any command is given
+ * and command is valid. Also validates the format of the dates given and ensures/enforces user compliance to
+ * input formats before parsing into an Executable
+ */
+public class Controller {
+    /**
+     * Object for executing the commands.
+     */
+    private final Executor exe;
+
+    private boolean isInUserAuthenticationMode = false;
+
+    /**
+     * Private constructor for a Parser object.
+     */
+    private Controller() {
+        exe = Executor.initExecutor();
+    }
+
+    /**
+     * Initializer for Parser.
+     *
+     * @return The Parser object.
+     */
+    public static Controller initController() {
+        return new Controller();
+    }
+
+    /**
+     * Returns a boolean to check if the user has a cached password.
+     *
+     * @return True if the user has already saved a password for Duke.
+     */
+    public boolean checkIsUserPwCached() {
+        Storage s = new Storage();
+        return s.checkPwCache();
+    }
+
+    /**
+     * Saves user entered password/name to the file.
+     *
+     * @param pw The userpassword/name.
+     */
+    public void savedUserPw(String pw) {
+        Storage s = new Storage();
+        s.saveUserPw(pw);
+    }
+
+
+    /**
+     * Parses given command and determines if it is a valid command,
+     * and calls an executor to execute a valid command.
+     *
+     * @param command
+     * @return Reply: what was done by the execution of input.
+     */
+    public String parseAndExec(String command) {
+        Parser parser = null;
+        if (!isInUserAuthenticationMode) {
+            try {
+                parser = Parser.parseAndCheck(command);
+            } catch (EmptyTaskException e) {
```

Comment:
Can consider removing the line breaks in the catch blocks as the code snippet in the coding standards for try-catch statements don't have the line breaks.